public/scripts/sysprompt.js

```diff
@@ -0,0 +1,236 @@
+        system_prompts = data.sysprompt;
+    }
+
+    migrateSystemPromptFromInstructMode();
+    toggleSystemPromptDisabledControls();
+
+    for (const prompt of system_prompts) {
+        $('<option>').val(prompt.name).text(prompt.name).appendTo($select);
+    }
+
+    $enabled.prop('checked', power_user.sysprompt.enabled);
+    $select.val(power_user.sysprompt.name);
+    $content.val(power_user.sysprompt.content);
+    if (!CSS.supports('field-sizing', 'content')) {
+        await resetScrollHeight($content);
+    }
+}
+
+/**
+ * Checks if the instruct template has a system prompt and prompts the user to save it as a system prompt.
+ * @param {string} name Name of the instruct template
+ * @param {object} template Instruct template object
+ */
+export async function checkForSystemPromptInInstructTemplate(name, template) {
+    if ('system_prompt' in template && name && template.system_prompt && !system_prompts.some(x => x.name === name)) {
+        const html = await renderTemplateAsync('migrateInstructPrompt', { prompt: template.system_prompt });
+        const confirm = await callGenericPopup(html, POPUP_TYPE.CONFIRM);
+        if (confirm) {
+            const prompt = { name: name, content: template.system_prompt };
+            const presetManager = getPresetManager('sysprompt');
+            await presetManager.savePreset(prompt.name, prompt);
+            toastr.success(`System prompt "${prompt.name}" has been saved.`);
+        } else {
+            toastr.info('System prompt has been discarded.');
+        }
+
+        delete template.system_prompt;
+    }
+}
+
+function toggleSystemPromptDisabledControls() {
+    $enabled.parent().find('i').toggleClass('toggleEnabled', !!power_user.sysprompt.enabled);
+    $contentBlock.toggleClass('disabled', !power_user.sysprompt.enabled);
+}
+
+function enableSystemPromptCallback() {
+    power_user.sysprompt.enabled = true;
+    $enabled.prop('checked', true);
+    toggleSystemPromptDisabledControls();
+    saveSettingsDebounced();
+    return '';
+}
+
+function disableSystemPromptCallback() {
+    power_user.sysprompt.enabled = false;
+    $enabled.prop('checked', false);
+    toggleSystemPromptDisabledControls();
+    saveSettingsDebounced();
+    return '';
+}
+
+function toggleSystemPromptCallback(_args, state) {
+    if (!state || typeof state !== 'string') {
+        return String(power_user.sysprompt.enabled);
+    }
+
+    const newState = isTrueBoolean(state);
+    newState ? enableSystemPromptCallback() : disableSystemPromptCallback();
+    return String(power_user.sysprompt.enabled);
+}
+
+function selectSystemPromptCallback(args, name) {
+    if (!power_user.sysprompt.enabled && !isTrueBoolean(args.forceGet)) {
+        return '';
+    }
+
+    if (!name) {
+        return power_user.sysprompt.name;
+    }
+
+    const quiet = isTrueBoolean(args?.quiet);
+    const instructNames = system_prompts.map(preset => preset.name);
+    const fuse = new Fuse(instructNames);
```

Comment:
Why are we using fuse here again? To allow typos when you type them by hand?

I mean I know this was the case in instruct and such too, but it leads to very weird scenarios. Doesn't have to be this PR, but we maybe have to rethink that.

![image](https://github.com/user-attachments/assets/74b80921-e88c-46f7-b11d-efe012edba66)
```
/let key=str Blinking | /sysprompt {{pipe}} | /echo Given: {{var::str}} Set: {{pipe}}
```

And there are worse *finds* it comes up with.

Reply:
I mean yeah, fallback searching might be okay. But she should likely document that at least, or people might be confused why setting "Blinking", that they didn't create yet, will select "Blank" :D